neural_compressor/torch/algorithms/weight_only/save_load.py

```diff
@@ -75,58 +82,81 @@ def load(model_name_or_path, original_model=None, format=LoadFormat.DEFAULT, dev
         self.format = format
         self.device = device
         self.kwargs = kwargs
         self.quantization_config = {}
-        self.loaded_state_dict_keys = {}
+        self.loaded_state_dict = {}
+        self.loaded_state_dict_keys = []
+        self._should_save_hpu_format_tensor = False
+        self._model_local_dir = None  # local directory where model files are saved
 
     def load_woq_model(self):
         if self.format == LoadFormat.HUGGINGFACE:
+            assert self.model_name_or_path is not None, "'model_name_or_path' can't be None."
+
             model = self.load_hf_format_woq_model()
             logger.info("Loading HuggingFace weight-only quantization model successfully.")
         elif self.format == LoadFormat.DEFAULT:
-            qmodel_weight_file_path = os.path.join(
-                os.path.abspath(os.path.expanduser(self.model_name_or_path)), WEIGHT_NAME
-            )
-            assert os.path.exists(qmodel_weight_file_path), (
-                "Cannot load model weight from path {}. "
-                "Please make sure '{}' file is saved in your '{}' directory ".format(
-                    qmodel_weight_file_path, WEIGHT_NAME, self.model_name_or_path
-                )
-            )
-
-            qconfig_file_path = os.path.join(os.path.abspath(os.path.expanduser(self.model_name_or_path)), QCONFIG_NAME)
-            assert os.path.exists(qconfig_file_path), (
-                "Cannot load model quantization config from path {}. "
-                "Please make sure '{}' file is saved in your '{}' directory".format(
-                    qconfig_file_path, QCONFIG_NAME, self.model_name_or_path
-                )
-            )
-
+            assert os.path.exists(self.model_name_or_path), f"'{self.model_name_or_path}' path doesn't exist."
             assert (
                 self.original_model is not None
             ), "Can't get original model. Please pass `original_model` to load function."
 
-            model = self.load_inc_format_woq_model(qmodel_weight_file_path, qconfig_file_path)
+            model = self.load_inc_format_woq_model()
             logger.info("Loading weight-only quantization model successfully.")
         else:
             raise ValueError(f"`format` in load function can only be 'huggingface' or 'default', but get {self.format}")
 
         return model
 
-    def load_inc_format_woq_model(self, qmodel_weight_file_path, qconfig_file_path):
-        qweights = torch.load(qmodel_weight_file_path)
-        self.loaded_state_dict_keys = qweights.keys()
+    def load_inc_format_woq_model(self):
+        self._model_local_dir = self.model_name_or_path
+
+        qmodel_weight_file_path = os.path.join(
+            os.path.abspath(os.path.expanduser(self.model_name_or_path)), WEIGHT_NAME
+        )
+        # if hpu format tensor can be used directly, then update qmodel_weight_file_path to the hpu format tensor file
+        if self._with_hpu_format_tensor():
```

Comment:
Good idea, we should use the format to indicate that the loaded model file is already in `hpu` format.
However, we think it is better to use the `habana` format because `hpu` is a device name. 
Looking forward to your feedback.

Reply:
Sorry, I made a mistake.  Thanks for correcting me,  @yuwenzho.
The arg: `format` indicates the load API arguments format, not packing format. We are using arg:  `device` to decide which packing format should be used. If the packing format is already hpu format, we should have some flag in config.json or else before uploading it to huggingface hub.
```python
# load API format 1
load(model_name_or_path="saved_results", original_model=fp32_model, format="default")
#  load API format 2
load(model_name_or_path="TheBloke/TinyLlama-1.1B-python-v0.1-GPTQ", format="huggingface")
```